lib/chat_api/billing.ex

```diff
@@ -34,6 +43,14 @@ defmodule ChatApi.Billing do
   # TODO: is this the proper way to handle this?
   # Basically, if the resource id is nil, we just want to return nil;
   # Otherwise, if the id exists, attempt to retrieve it
+  @spec retrieve_stripe_resource(
+          atom(),
+          nil | binary | Stripe.Product.t() | Stripe.Subscription.t() | Stripe.PaymentMethod.t()
```

Comment:
should this just be `nil | binary`?

Reply:
Based on code it can move to: 
```elixir
Stripe.Subscription.retrieve(subscription_id)
Stripe.Product.retrieve(product_id)
Stripe.PaymentMethod.retrieve(payment_method_id)
```

And they are returning:
```elixir
Stripe.Subscription.t()
Stripe.Product.t()
Stripe.PaymentMethod.t()
```

Accordingly. For me looks like this is a good spec 😄

Reply:
Sorry, saw a underlaying code, in account it always be a `binary()` will fix it